src/Util/SortByFilters.ts

```diff
@@ -1,10 +1,10 @@
 type SortConfig<T> = {
   getValue: (id: T) => number | string
-  tieBreaker?: string
+  tieBreakers?: string[]
 }
 export type SortConfigs<Keys extends string, T> = Record<Keys, SortConfig<T>>
 
-export function sortFunction<Keys extends string, T>(sortby: string, ascending: boolean, configs: SortConfigs<Keys, T>) {
+export function sortFunction<Keys extends string, T>(sortbyKeys: string[], ascending: boolean, configs: SortConfigs<Keys, T>): (a: T, b: T) => number {
```

Comment:
The `sortByKeys` can be determined by user input (aka what they select for the sorting criteria, whereas the `tiebreakers` are defined per-key by us. So in this case, I sort with `sortByKeys = ["favorite", userSelectedSortKey]`, so that way favorites are first, and then subsequently sorted by the `userSelectedSortKey`

Reply:
The `sortByKeys` can be determined by user input (aka what they select for the sorting criteria, whereas the tiebreakers are defined per-key by us. So in this case, I sort with `sortByKeys = ["favorite", userSelectedSortKey]`, so that way favorites are first, and then subsequently sorted by the `userSelectedSortKey`.

Additionally, I want the order of sorting to be `["favorite", favorite.tiebreakers, userSelectedSortKey, userSelectedSortKey.tiebreakers]`, which I could implement so callers of `sortFunction` would have to then pass in the tiebreakers, similary like `["favorite", ...sortConfigs.favorite.tiebreakers, userSelectedSortKey, ...sortConfigs[userSelectedSortKey].tiebreakers]`, which feels messier than just handling the tiebreakers in `sortFunction`

Reply:
How does it look now?